File: 1894-find-the-student-that-will-replace-the-chalk/1894-find-the-student-that-will-replace-the-chalk.py
class Solution:
    def chalkReplacer(self, chalk: List[int], k: int) -> int:
        tot=sum(chalk)
        k%=tot
        for i in range(len(chalk)):
            if k<chalk[i]:
                return i
            k-=chalk[i]
        return 0
        

File: 1894-find-the-student-that-will-replace-the-chalk/1894-find-the-student-that-will-replace-the-chalk.java
class Solution {
    public int chalkReplacer(int[] chalk, int k) {
        int len=chalk.length;
        long tot=0;
        for(int i: chalk){
            tot+=i;
        }
        k%=tot;
        for(int i=0;i<len;i++){
            if(k<chalk[i]){
                return i;
            }
            k-=chalk[i];
        }
        return 0;
    }
}